src/Ideas.js

```diff
@@ -8,91 +8,116 @@ import './Ideas.css';
     const query = new URLSearchParams(location.search);
-    const value = query.get('categories') || '';
+    const queryCategories = query.get('categories') || '';
+    const queryTags = query.get('tags') || '';
+
+    allValues.categories = allCategories;
+    allValues.tags = unique(ideas.reduce((a, b) => a.concat(b.tags), []).filter(Boolean));
 
-    const categories = value ? value.split(',') : defaultSelection;
+    const categories = queryCategories ? queryCategories.split(',') : allValues.categories;
+    const tags = queryTags ? queryTags.split(',') : allValues.tags;
 
-    this.state = { categories, search: '' };
+    this.state = {
+      categories,
+      tags,
+      search: '',
+    };
   }
 
   componentWillReceiveProps(nextProps) {
     const queryParams = new URLSearchParams(nextProps.location.search);
 
     if (!queryParams.get('categories')) {
-      this.setState({
-        categories: defaultSelection,
-      });
+      this.setState({ categories: allValues.categories });
     }
   }
 
   changeSearch = (event) => {
     this.setState({ search: event.target.value });
   }
 
-  changeCategory = (clickedCategory) => {
-    let categories = this.state.categories.slice();
-    const { history } = this.props;
+  // updates filter state and query params based on a
+  // clicked value, list of all possible values, and type
+  changeFilter = (value, all, type) => {
+    let currentSelected = this.state[type].slice();
 
-    const allWereSelected = checkIfAllSelected(categories, defaultSelection);
+    const allWereSelected = checkIfAllSelected(currentSelected, all);
 
     if (allWereSelected) {
-      categories = [clickedCategory];
-    } else if (categories.indexOf(clickedCategory) > -1) {
-      categories = removeItem(categories, clickedCategory);
+      currentSelected = [value];
+    } else if (currentSelected.indexOf(value) > -1) {
+      currentSelected = removeItem(currentSelected, value);
     } else {
-      categories.push(clickedCategory);
+      currentSelected.push(value);
     }
 
-    if (categories.length === 0) categories = defaultSelection;
+    if (currentSelected.length === 0) currentSelected = all;
+
+    this.state[type] = currentSelected;
+    this.forceUpdate();
 
-    this.setState({ categories });
+    this.updateParams();
+  }
+
+  // updates url query params from state
+  updateParams() {
+    const { history } = this.props;
+    const { categories, tags } = this.state;
 
-    const allAreSelected = checkIfAllSelected(categories, defaultSelection);
+    const paramChunks = [];
 
-    history.push({
-      search: allAreSelected ? '' : `?categories=${categories.join(',')}`,
-    });
+    if (!checkIfAllSelected(categories, allValues.categories)) paramChunks.push(`categories=${categories.join(',')}`);
+    if (!checkIfAllSelected(tags, allValues.tags)) paramChunks.push(`tags=${tags.join(',')}`);
+
+    const search = (paramChunks.length === 0) ? '' : `?${paramChunks.join('&')}`;
```

Comment:
leaving for now, maybe a package can help us with the state <-> query params thing later